infrastructure/src/main/java/mops/database/dto/VeranstaltungDto.java

```diff
@@ -0,0 +1,45 @@
+package mops.database.dto;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.relational.core.mapping.Column;
+import org.springframework.data.relational.core.mapping.Table;
+
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@AllArgsConstructor
+@Data
+@Table("veranstaltung")
+public class VeranstaltungDto {
+
+  @Id
+  Long id;
+  String name;
+  Integer semester;
+  @Column("veranstaltung")
+  Set<FragebogenDto> frageboegen;
+  @Column("veranstaltung")
+  Set<SBelegtVDto> studenten;
+  @Column("veranstaltung")
+  Set<DOrganisiertVDto> dozenten;
+
+
+  public static VeranstaltungDto create(String name, Integer semester) {
+    return new VeranstaltungDto(null, name, semester, new HashSet<>(),
+        new HashSet<>(), new HashSet<>());
+  }
+
+  public void addStudent(StudentDto student) {
+    studenten.add(new SBelegtVDto(student.getId()));
+  }
+
+  public void addDozent(DozentDto dozent) {
+    dozenten.add(new DOrganisiertVDto(dozent.getId()));
+  }
+
+  public void addFragebogen(FragebogenDto fragebogen) {
+    frageboegen.add(fragebogen);
+  }
```

Comment:
Same as in a few classes above

Reply:
addStudent & addDozent are necessary because the methods are used to implement and save a n:m relationship in another table.